agent/consul/rpc.go

```diff
@@ -1043,3 +1048,31 @@ func (s *Server) consistentRead() error {
 
 	return structs.ErrNotReadyForConsistentReads
 }
+
+// maskResultsFilteredByACLs blanks out the ResultsFilteredByACLs flag if the
+// request is unauthenticated, to limit information leaking.
+//
+// Endpoints that support bexpr filtering could be used in combination with
+// this flag/header to discover the existence of resources to which the user
+// does not have access, therefore we only expose it when the user presents
+// a valid ACL token. This doesn't completely remove the risk (by nature the
+// purpose of this flag is to let the user know there are resources they can
+// not access) but it prevents completely unauthenticated users from doing so.
+//
+// Notes:
+//
+//	* The definition of "unauthenticated" here is incomplete, as it doesn't
+//	  account for the fact that operators can modify the anonymous token with
+//	  custom policies, or set namespace default policies. As these scenarios
+//	  are less common and this flag is a best-effort UX improvement, we think
+//	  the trade-off for reduced complexity is acceptable.
+//
+//	* This method assumes that the given token has already been validated (and
+//	  will only check whether it is blank or not). It's a safe assumption because
+//	  ResultsFilteredByACLs is only set to try when applying the already-resolved
+//	  token's policies.
+func (s *Server) maskResultsFilteredByACLs(token string, meta structs.QueryMetaCompat) {
```

Comment:
Is there any particular reason for making this a method off `(s *Server)`?

Reply:
Nah, not really. Do you have a preference for making it a bare function instead?

Reply:
I'd personally prefer it be a bare function since methods add to noise if I ever try to autocomplete something on (*Server), but I think the signature also misleads you into thinking the server is somehow involved in the function.

Reply:
Cool, makes sense to me. Have made it a function 🙇🏻